### PR TITLE
Update runtime, grab latest git commit.

### PR DESCRIPTION
**Pull request by F-i-f**
Update runtime to 23.08.
Update to latest git commit.
Drop SDL2 patch upstreamed.
Add C++ compilation fix for 22.08/23.08 SDKs.

### CODE REVIEW

File: net.sourceforge.DuneLegacy.yaml [hunk 1]
```diff
@@ -24,13 +24,12 @@ modules:
     sources:
       - type: git
         url: https://git.code.sf.net/p/dunelegacy/code
-        tag: v0.96.4
-        commit: dede956fbd6f6f6bbee98a059c7edb677e67fc6d
+        commit: 6ea9ac96854daa8c75ba429e78dc6716b147e106
```
Comment (Mailaender) [inline net.sourceforge.DuneLegacy.yaml]: We need a released version. Ask upstream for a non-Windows release. https://dunelegacy.sourceforge.net/website/news.html#2022-12-27-00-00-00

**Comment by F-i-f** [comment]
The project seems more or less abandoned, so I'm not sure I'll be able to get a release.
Would you accept a PR that:
- Fixes the linter errors
- Updates the runtime to 23.08
- But keeps the release at 0.96.4?

**Comment by Mailaender** [comment]
Yes, that would be very much appreciated.

**Comment by Mailaender** [comment]
bot, build

**Comment by F-i-f** [comment]
Thank you for taking care of restarting the (transiently broken) build and merging.

**Pull request merged**